tests/scala/org/pantsbuild/zinc/compiler/JarCreationSpec.scala

```diff
@@ -0,0 +1,41 @@
+package org.pantsbuild.zinc.compiler
+
+import java.io.File
+import java.nio.file.{Path, Paths}
+import scala.collection.mutable
+
+import sbt.io.IO
+
+import org.junit.runner.RunWith
+import org.scalatest.WordSpec
+import org.scalatest.junit.JUnitRunner
+import org.scalatest.MustMatchers
+
+@RunWith(classOf[JUnitRunner])
+class JarCreationSpec extends WordSpec with MustMatchers {
+  "JarCreationWithoutClasses" should {
+    "succeed when input classes are provided" in {
+      IO.withTemporaryDirectory { tempDir =>
+        val filePaths = new mutable.TreeSet[Path]()
+        val jarOutputPath = Paths.get(tempDir.toString, "spec-empty-output.jar")
+        OutputUtils.createJar(filePaths, jarOutputPath, System.currentTimeMillis())
+        OutputUtils.existsClass(jarOutputPath, "NonExistent.class") must be(false)
+      }
+    }
+  }
+  "JarCreationWithClasses" should {
+    "succeed when input classes are provided" in {
+      IO.withTemporaryDirectory { tempDir =>
```

Comment:
I think that the code and this test assume that the jar is always created under the classes directory... but that won't be the case.

Can you change this test to use two temporary directories: one containing the input classes, and another containing the output jar?